### PR TITLE
Path grouping strategy

### PR DESCRIPTION
**Pull request by Saibamen**
Users can write function to generate path grouping.

For example, I can write path with `/myPath?queryParam1?queryParam2`, so I will not have `SwaggerGeneratorException` (`Conflicting method/path combination x for actions (...)`) when mapping old .NET Framework API with `Route("/api/[controller]")`.

Maybe related to https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues/2354, but I don't know if `ApiDescription` has information about `ConsumesAttribute`.

### CODE REVIEW

File: src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGeneratorOptions.cs [hunk 1]
```diff
@@ -49,6 +50,8 @@ public SwaggerGeneratorOptions()
 
         public Func<ApiDescription, string> SortKeySelector { get; set; }
 
+        public Func<ApiDescription, string> PathGroupingStrategy { get; set; }
```
Comment (martincostello) [inline src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGeneratorOptions.cs:53]: What about `PathGroupSelector` for consistency? I'm not a fan of "strategy".

File: test/Swashbuckle.AspNetCore.SwaggerGen.Test/ConfigureSwaggerGeneratorOptionsTests.cs [hunk 2]
```diff
@@ -18,7 +18,7 @@ public static void DeepCopy_Copies_All_Properties()
 
         // If this assertion fails, it means that a new property has been added
         // to SwaggerGeneratorOptions and ConfigureSwaggerGeneratorOptions.DeepCopy() needs to be updated
```
Comment (martincostello) [inline test/Swashbuckle.AspNetCore.SwaggerGen.Test/ConfigureSwaggerGeneratorOptionsTests.cs:20]: This hasn't been done.

Comment (Saibamen) [reply test/Swashbuckle.AspNetCore.SwaggerGen.Test/ConfigureSwaggerGeneratorOptionsTests.cs:20]: Done

**Review by martincostello — APPROVED** [review]
Could you change this to 7.2.0 please?

https://github.com/domaindrivendev/Swashbuckle.AspNetCore/blob/1715165d85f353e3a14d6878a99e2c51a58919c6/Directory.Build.props#L45

**Comment by Saibamen** [comment]
Wait. I will change exception message

**Comment by martincostello** [comment]
I was just about to enable auto-merge on this - what are you changing?

File: src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs [hunk 3]
```diff
@@ -284,7 +284,7 @@ private async Task<Dictionary<OperationType, OpenApiOperation>> GenerateOperatio
             {
                 throw new SwaggerGeneratorException(string.Format(
                     "Conflicting method/path combination \"{0} {1}\" for actions - {2}. " +
-                    "Actions require a unique method/path combination for Swagger/OpenAPI 3.0. Use ConflictingActionsResolver as a workaround",
+                    "Actions require a unique method/path combination for Swagger/OpenAPI 2.0 and 3.0. Use ConflictingActionsResolver as a workaround or write your own PathGroupSelector",
```
Comment (martincostello) [inline src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs:287]:
```suggestion
                    "Actions require a unique method/path combination for Swagger/OpenAPI 2.0 and 3.0. Use ConflictingActionsResolver as a workaround or provide your own implementation of PathGroupSelector.",
```

Comment (Saibamen) [inline src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs:287]: OpenAPI 2.0 specs are not clear for this like in 3.0:

image


But I look back in Swashbuckle code. The same IF statement was there in 2015. OpenAPI 3.0 was released in 2017.

image

**Pull request merged**